Services/DeviceModels.cs

```diff
@@ -92,5 +94,50 @@ private List<string> GetDeviceModelFiles()
 
             return this.deviceModelFiles;
         }
+
+        public HashSet<string> GetPropNames()
+        {
+            var list = this.GetList().ToList();
+            if (list.Count > 0)
+            {
+                var set = new HashSet<string>();
+                list.ForEach(m =>
+                {
+                    foreach (var item in m.Properties)
+                    {
+                        this.PreparePropNames(set, item.Value, item.Key);
+                    }
+                });
+                var result = new HashSet<string>();
+                foreach (string prop in set)
+                {
+                    result.Add("Properties.Reported." + prop);
+                }
+                return result;
+            }
+
+            return null;
+        }
+
+        private void PreparePropNames(HashSet<string> set, object obj, string prefix)
+        {
+            if (obj is JValue)
+            {
+                set.Add(prefix);
+                return;
+            }
+
+            if (obj is bool || obj is string || double.TryParse(obj.ToString(), out _))
+            {
+                set.Add(prefix);
+                return;
+            }
+
+            foreach (var item in (obj as JToken).Values())
+            {
+                var path = item.Path;
+                this.PreparePropNames(set, item, $"{prefix}.{(path.Contains(".") ? path.Substring(path.LastIndexOf('.') + 1) : path)}");
```

Comment:
Could you add a comment explaining what format this needs to be in and why?

Reply:
done